backend/LexBoxApi/Services/HgService.cs

```diff
@@ -80,32 +86,24 @@ public async Task PrepareEmptyRepo(string code, string tempRepoSuffix)
         var tempRepoName = $"{code}__{tempRepoSuffix}";
         await Task.Run(() =>
         {
-            var deletedRepoPath = Path.Combine(_options.Value.RepoPath, DELETED_REPO_FOLDER);
-            var directory = Directory.CreateDirectory(deletedRepoPath);
+            var tempRepoPath = Path.Combine(_options.Value.RepoPath, TEMP_REPO_FOLDER);
+            var directory = Directory.CreateDirectory(tempRepoPath);
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
                 directory.UnixFileMode = Permissions;
             var dest = Directory.CreateDirectory(Path.Combine(directory.FullName, tempRepoName));
```

Comment:
I think creating the directory should be in `InitRepoAt` as well.

Reply:
Done in commit af9dcae9.

Reply:
It looks like my comment in https://github.com/sillsdev/languageforge-lexbox/pull/740#discussion_r1575872486 was too broad and my comment here too specific.

But, there's still code that is duplicated here and in `InitRepoForCode`. Please factor out the duplication.

Reply:
I refactored this away. @myieye might want to look again